core/parameters/src/cost.rs

```diff
@@ -359,6 +359,9 @@ pub struct RuntimeFeesConfig {
 
     /// Pessimistic gas price inflation ratio.
     pub pessimistic_gas_price_inflation_ratio: Rational32,
+
+    /// The maximum size of the state witness after which we defer execution of any new receipts.
+    pub state_witness_size_soft_limit: usize,
```

Comment:
Is there a reason this is not a part of the `LimitConfig`? It is a limit after all.

If its because `LimitConfig` itself is a part of `vm::Config`, then I think we might want to "just" move the `LimitConfig` such that it is no longer as specific to VM. On the other hand, it doesn't matter either way: we can easily change the place of the parameter in the future so long as we take care to keep the RuntimeConfigView backwards compatible.

Reply:
Alternatively you could move it to `RuntimeConfig` proper.

Reply:
In the future, I anticipate more stateless validation related configs like state witness hard_limit for individual contracts etc. to come in. I personally wasn't sure whether I should create a new struct like `StatelessValidationConfig` in `RuntimeConfig` or just put it in `RuntimeFeesConfig`.

Yes, I don't like `vm::Config` for this as it seems specific to VMs and `state_witness_size_soft_limit` is technically used outside the VM.

Not too sure what do do here, but like you suggested, let me try to move this to `RuntimeConfig` proper.